bundles/binding/org.openhab.binding.cardio2e/src/main/java/org/openhab/binding/cardio2e/internal/Cardio2eBinding.java

```diff
@@ -0,0 +1,944 @@
+	 * @param reason Reason code for the deactivation:<br>
+	 * <ul>
+	 * <li> 0 – Unspecified
+     * <li> 1 – The component was disabled
+     * <li> 2 – A reference became unsatisfied
+     * <li> 3 – A configuration was changed
+     * <li> 4 – A configuration was deleted
+     * <li> 5 – The component was disposed
+     * <li> 6 – The bundle was stopped
+     * </ul>
+	 */
+	
+	public void deactivate(final int reason) {
+		this.bundleContext = null;
+		// deallocate resources here that are no longer needed and 
+		// should be reset when activating this binding again
+		logger.debug("deactivate");
+		purgeCardio2eBinding();
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected long getRefreshInterval() {
+		return refreshInterval;
+	}
+
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected String getName() {
+		return "Cardio2e Refresh Service";
+	}
+	
+	/**
+	 * @{inheritDoc}
+	 */
+	@Override
+	protected void execute() {
+		// the frequently executed code (polling) goes here ...
+		logger.debug("execute() method is called!");
+		if (loggedIn) {
+			//Cardio 2é date and time cyclic ask code used for progressive update only
+			if ((datetimeMaxOffset >= 0) && (pendingDateTimeUpdate.getDateTime() != null)) {
+            	getCardioDateTime();
+            }
+		}
+	}
+
+	public void initializeCardio2eBinding(final Map<String, Object> configuration) {
+		int minDelayBetweenReceivingAndSending = 0;
+		int minDelayBetweenSendings = 0;
+		// Set default values 
+		port = null;
+		programCode = "00000";
+		securityCode = null;
+		zoneStateDetection = false;
+		zoneUnchangedMinRefreshDelay = 600000;
+		datetimeMaxOffset = 15;
+		firstUpdateWillSetDatetime = false;
+		allowedDatetimeUpdateHour = -1;
+		testMode = false;
+		Cardio2eTransactionParser.filterUnnecessaryCommand = false;
+		Cardio2eTransactionParser.filterUnnecessaryReverseModeUpdate = true;
+		Cardio2eLightingTransaction.setSmartSendingEnabledClass (false);
+		Cardio2eRelayTransaction.setSmartSendingEnabledClass (false);
+		Cardio2eHvacControlTransaction.setSmartSendingEnabledClass (false);
+		Cardio2eDateTimeTransaction.setSmartSendingEnabledClass (false);
+		Cardio2eScenarioTransaction.setSmartSendingEnabledClass (false);
+		Cardio2eSecurityTransaction.setSmartSendingEnabledClass (false);
+		Cardio2eZonesBypassTransaction.setSmartSendingEnabledClass (false);
+		Cardio2eCurtainTransaction.setSmartSendingEnabledClass (false);
+		//Load config values
+		String portString = (String) configuration.get("port");
+		if (StringUtils.isNotBlank(portString)) {
+			port = portString;
+            logger.debug("Serial port set from config file: " + port);
+		}
+		if (port != null){
+			String programcodeString = (String) configuration.get("programcode");
+			if (StringUtils.isNotBlank(programcodeString)) {
+				programCode = programcodeString;
+	            logger.debug("Program code updated from config file");
```

Comment:
The omission of the value is intentional, so as not to show the programming code in the log.